rust/examples/otel-stdout.rs

```diff
@@ -46,13 +45,8 @@ pub async fn main() -> anyhow::Result<()> {
         .get_func(&mut store, function_name)
         .expect("function exists");
 
-    OtelStdoutAdapter::start_trace(
-        String::from("module-name"),
-        function_name.to_string(),
-        || {
-            f.call(&mut store, &[], &mut []).unwrap();
-        },
-    );
+    OtelStdoutAdapter::set_trace_id(&collector, "any-old-trace-id".to_string()).await;
```

Comment:
This is a little weird because the collector is trying to be agnostic to the adapter but trace-id is an adapter concern. Do you think every adapter is going to have a trace-like-thing enough that we could create a `set_trace_id` on the collector?

Reply:
I'm looking at the API a bit and seeing if we can get rid of some of the public-facing collector stuff, which may eliminate this concern.